app/queries/keywords_query.rb

```diff
@@ -1,43 +1,41 @@
 # frozen_string_literal: true
 
 class KeywordsQuery
-  attr_reader :scope, :filter_params
-
   def initialize(scope, filter_params)
     @scope = scope
-    @filter_params = filter_params
+    @filter = filter_params
   end
 
   def call
-    return filter_ads_top_urls if filter_params[:adwords_url_contains]
-
-    return filter_result_urls if filter_params[:word] && filter_params[:match_at_least]
-
-    scope
-  rescue NoMethodError
-    scope
+    filter_urls
   end
 
   private
 
-  def filter_ads_top_urls
-    word = filter_params[:adwords_url_contains]&.downcase
+  attr_reader :scope, :filter
+  attr_accessor :keywords
 
-    return scope unless word
+  def filter_urls
+    return scope unless filter
 
-    scope.where("array_to_string(ads_top_urls, '||') ILIKE ?", "%#{word}%")
-  end
+    @keywords = scope
 
-  def filter_result_urls
-    word = filter_params[:word]
-    number_of_matches = filter_params[:match_at_least]
+    @keywords = filter_ads_top_urls if filter[:adwords_url_contains].present?
+    @keywords = filter_result_urls if filter[:result_url_contains].present?
+    @keywords = filter_match_at_least if filter[:word].present? && filter[:match_at_least].present?
```

Comment:
```suggestion
    scope = filter_ads_top_urls if filter[:adwords_url_contains].present?
    scope = filter_result_urls if filter[:result_url_contains].present?
    scope = filter_match_at_least if filter[:word].present? && filter[:match_at_least].present?
```

Is there a reason we don't use `scope` everywhere? 👀

Reply:
Good catch, I first worry to change `scope` so I created `keywords`. Will revert this.

Reply:
Fixed in https://github.com/nimblehq/ic-rails-phong-bliss/pull/78/commits/a68bd8109b511dc26a8f3ac670e0c417ed051ebc